### PR TITLE
Big update

### PR DESCRIPTION
**Pull request by alexreinking**
Trying to turn this into a real project that plays nicely with Python and doesn't have to be a SYSTL submodule. I also renamed the module from `adt` to `asdl_adt`, which isn't taken on PyPI.

### CODE REVIEW

File: src/adt/adt.py [hunk 1]
```diff
@@ -297,42 +295,52 @@ def _add_memoization(mod, whitelist, ext_key):
...
-        i = 'i' if f.name != 'i' else 'ii'
-        return (f"tuple( K['{f.type}']({i}) "
-                f"for {i} in {f.name} ),")
+        i = "i" if f.name != "i" else "ii"
+        return f"tuple( K['{f.type}']({i}) " f"for {i} in {f.name} ),"
 
     def create_optkey(f):
-        return (f"None if {f.name} == None else "
-                f"K['{f.type}']({f.name}),")
+        return f"None if {f.name} == None else " f"K['{f.type}']({f.name}),"
```
Comment (gilbo) [inline src/adt/adt.py]: This should probably be `is None`.  Bug from back when I was confused about `is` vs `==`

Comment (gilbo) [reply src/adt/adt.py]: e.g. consider a `bool?`

File: src/adt/adt.py [hunk 2]
```diff
@@ -76,95 +70,94 @@ def basic_check(i, name, argname, typ, indent="    "):
...
-        return exec_out[C_name + '_repr']
+        return exec_out[C_name + "_repr"]
 
     def create_eqfn(C_name, fields):
-        compares = ' and '.join(
-            ['type(self) == type(o)'] +
-            [f"(self.{f.name} == o.{f.name})" for f in fields]
+        compares = " and ".join(
+            ["type(self) == type(o)"]
+            + [f"(self.{f.name} == o.{f.name})" for f in fields]
```
Comment (gilbo) [inline src/adt/adt.py:142]: As mentioned below, should this be `is` instead of `==`?

File: src/adt/adt.py (hunk 1 above)
Comment (alexreinking) [reply src/adt/adt.py]: Already changed to `is None` :)

File: src/adt/adt.py (hunk 2 above)
Comment (alexreinking) [reply src/adt/adt.py:142]: `is` for the `type` comparison, most likely.

**Pull request merged**